css-shared-element-transitions/index.bs

```diff
@@ -298,241 +299,299 @@ Single-page API:
+            * |element| is not |element|'s [=tree/root=] and |element| does prevent [=fragmentation=].
+
+            Then set |preparationFailed| to true and [=break=].
+
+        1. [=set/Append=] |transitionTag| to |usedTransitionTags|.
+
+        1. Let |capture| be a new [=captured element=] struct.
+
+        1. Set |capture|'s [=outgoing image=] to the result of [=capturing the image=] of |element|.
+
+        1. Set |capture|'s [=outgoing styles=] to the following:
+
+            : 'transform'
+            :: A CSS transform that would place |element|
+                from the document origin to its current quad.
+            :: This value is identity for the root element.
+
+            : 'width'
+            : 'height'
+            :: The width and height of |element|'s border box.
+            :: This value is the bounds of the initial containing block for the root element.
+
+            : 'object-view-box'
+            :: An 'object-view-box' value that,
+                when applied to the outgoing image,
+                will cause the view box to coincide with |element|'s [=border box=]
+                in the image.
+
+            : 'writing-mode'
+            :: The 'writing-mode' of |element|.
+
+            : 'direction'
+            :: The 'direction' of |element|.
+
+            Issue: This needs proper types.
+
+        1. Set |taggedElements|[|transitionTag|] to |capture|.
+
+    1. Suppress rendering opportunities for |transition|'s [=relevant global object's=] [=associated document=].
+
+        Note: The aim is to prevent unintended DOM updates from being presented to the
+        user after a cached snapshot for the elements has been captured. We wait for
+        one rendering opportunity after prepare to present DOM mutations made by the
+        author before prepare to be presented to the user. This is also the content
+        captured in snapshots.
+
+        Issue: Further clarify this behavior. Lifecycle updates can still be triggered
+        via script APIs which query style or layout information but no visual updates
+        are presented to the user. Is this the same behavior as render-blocking?
+
+        Issue: How should input be handled when in this state? The last frame presented
+        to the user will not reflect the DOM state as it asynchronously switches to
+        the new version.
+
+    1. [=Queue a global task=] on the [=DOM manipulation task source=],
+        given |transition|'s [=relevant global object=],
+        to execute the following steps:
+
+            Note: A task is queued here because the texture read back in [=capturing the image=] may be async,
+                although the render steps in the HTML spec act as if it's synchronous.
+
+        1. Let |timedOut| be false.
+
+        1. [=/Invoke=] [=SameDocumentTransition/prepare callback=],
+            and let |userP| be the return value.
+
+        If |userP| does not settle within an implementation-defined timeout period:
```

Comment:
It's a callback rather than a step, like "[Upon fulfillment](https://webidl.spec.whatwg.org/#upon-fulfillment) of userP:". It's a bit hand-wavy, but I'm not sure of a better way.